ray-curator/ray_curator/stages/audio/inference/asr_nemo.py

```diff
@@ -0,0 +1,145 @@
+import time
+from dataclasses import dataclass, field
+from typing import ClassVar
+
+import nemo.collections.asr as nemo_asr
+import torch
+from loguru import logger
+
+from ray_curator.backends.base import NodeInfo, WorkerMetadata
+from ray_curator.stages.base import ProcessingStage
+from ray_curator.stages.resources import Resources
+from ray_curator.tasks import DocumentBatch, FileGroupTask, SpeechObject
+
+
+@dataclass
+class InferenceAsrNemoStage(ProcessingStage[FileGroupTask | DocumentBatch | SpeechObject, SpeechObject]):
+    """Stage that do speech recognition inference using NeMo model.
+
+    Args:
+        model_name (str): name of the speech recognition NeMo model. See full list at https://docs.nvidia.com/nemo-framework/user-guide/latest/nemotoolkit/asr/all_chkpt.html
+        filepath_key (str): which key of the data object should be used to find the path to audiofile. Defaults to “audio_filepath”
+        pred_text_key (str): key is used to identify the field containing the predicted transcription associated with a particular audio sample. Defaults to “pred_text”
+        cuda (str): device to run inference on it. Could be cpu, cuda, ipu, xpu, mkldnn, opengl, opencl, ideep, hip, ve, fpga, maia, xla, lazy, vulkan, mps, meta, hpu, mtia or cuda number (digit). Defaults to “” (empty string)
```

Comment:
```suggestion
        cuda (str): device to run inference on it. Could be cpu or cuda.
```
Would prefer not to allow the user to set a GPU number, etc. Also, I think having the `cuda` parameter and the `.with(Resources(...))` is unnecessary, just 1 should be needed. How about using `with`, and if GPUs > 0, then set the `map_location` accordingly?